cmd/up/space/prerequisites/opentelemetrycollector/opentelemetrycollector.go

```
@@ -0,0 +1,129 @@
+	crdclient *apixv1client.ApiextensionsV1Client
+	kclient   kubernetes.Interface
+}
+
+// New constructs a new OpenTelemetryCollectorMgr instance that can used to install the
+// opentelemetry-operator chart.
+func New(config *rest.Config) (*OpenTelemetryCollectorOperator, error) {
+	mgr, err := helm.NewManager(config,
+		chartName,
+		otelMgrURL,
+		helm.WithNamespace(chartName),
+	)
+	if err != nil {
+		return nil, errors.Wrap(err, fmt.Sprintf(errFmtCreateHelmManager, chartName))
+	}
+	crdclient, err := apixv1client.NewForConfig(config)
+	if err != nil {
+		return nil, errors.Wrap(err, fmt.Sprintf(errFmtCreateK8sClient, chartName))
+	}
+	kclient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, errors.Wrap(err, fmt.Sprintf(errFmtCreateK8sClient, chartName))
+	}
+
+	return &OpenTelemetryCollectorOperator{
+		mgr:       mgr,
+		crdclient: crdclient,
+		kclient:   kclient,
+	}, nil
+}
+
+// GetName returns the name of the opentelemetry-operator chart.
+func (c *OpenTelemetryCollectorOperator) GetName() string {
+	return chartName
+}
+
+// Install performs a Helm install of the chart.
+func (c *OpenTelemetryCollectorOperator) Install() error {
+	if c.IsInstalled() {
+		// nothing to do
+		return nil
+	}
+
+	// create namespace before creating chart.
+	_, err := c.kclient.CoreV1().
+		Namespaces().
+		Create(context.Background(),
+			&corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: chartName,
+				},
+			}, metav1.CreateOptions{})
+	if err != nil && !kerrors.IsAlreadyExists(err) {
+		return errors.Wrap(err, fmt.Sprintf(errFmtCreateNamespace, chartName))
+	}
+
+	return c.mgr.Install(version, values)
+}
+
+// IsInstalled checks if opentelemetry operator has been installed in the target cluster.
+func (c *OpenTelemetryCollectorOperator) IsInstalled() bool {
+	_, err := c.crdclient.
+		CustomResourceDefinitions().
+		Get(
+			context.Background(),
+			otelCollectorCRD,
+			metav1.GetOptions{},
+		)
+	return !kerrors.IsNotFound(err)
```

This is what we end up having:

err == nil => Installed
err == NotFound => Not installed
err == any other error => Installed

The last one is not correct. We should simply return error in that case instead of assuming it is installed or not.